tests/legacy/unit/store/test_errors.py

```diff
@@ -113,3 +116,186 @@ def test_snapcraft_exception_handling(
         assert exception.get_details() == expected_details
         assert exception.get_docs_url() == expected_docs_url
         assert exception.get_reportable() == expected_reportable
+
+
+class TestErrorFormatting:
+
+    scenarios = [
```

Comment:
It wasn't necessary to add tests for the other `storeapi` errors, but I see these tests for errors elsewhere in snapcraft, so I decided to add them here too.